packages/hub/src/components/NavigationBlocker/context.ts

```diff
@@ -0,0 +1,69 @@
+import { createContext, Reducer } from "react";
+
+type State = {
+  // Blockers are functions; this way we can check dynamic parameters such as
+  // form values to decide whether navigation should be blocked.
+  blockers: Record<string, () => boolean>;
+  // Link was intercepted, need to show a modal.
+  interceptedLink: string | undefined;
+};
+
+type Action =
+  | {
+      type: "addBlocker";
```

Comment:
not important, but if I were just looking at this word, I'd assume it meant 'ad blocker'